### PR TITLE
BUG: Fix millisecond issue for omniscidb, mysql, postgresql and pandas

### LINKED ISSUE #2166: BUG: [PostgreSQL] Millisecond extraction rounds up the result
Currently the millisecond extraction is rounding up the result. 

Using the same approach used by pandas (for example for microsecond), it should use a `floor` method to round it down.

### LINKED ISSUE #2167: BUG: OmniSciDB Millisecond extraction doesn't work
OmniSciDB Millisecond doesn't work, probably because it is using the Impala translation.

```
E           Exception: Exception: Parse failed: Encountered "\"timestamp_col\"" at line 1, column 16.
E           Was expecting one of:
E               "CENTURY" ...
E               "DAY" ...
E               "DECADE" ...
E               "DOW" ...
E               "DOY" ...
E               "EPOCH" ...
E               "HOUR" ...
E               "ISODOW" ...
E               "ISOYEAR" ...
E               "MICROSECOND" ...
E               "MILLISECOND" ...
E               "MILLENNIUM" ...
E               "MINUTE" ...
E               "MONTH" ...
E               "NANOSECOND" ...
E               "QUARTER" ...
E               "SECOND" ...
E               "WEEK" ...
E               "YEAR" ...
E               : SELECT extract("timestamp_col", 'millisecond') AS tmp
E           FROM functional_alltypes

ibis/omniscidb/client.py:845: Exception
```

### LINKED ISSUE #2168: BUG: Pandas Millisecond extraction doesn't work
Pandas backend uses the follow expression for ExtractTemporalField:

`return getattr(data.dt, field_name).astype(np.int32)`

as `dt` doesn't have `millisecond` it raises an error.

### LINKED ISSUE #2169: BUG: MySQL Millisecond extraction doesn't work
MySQL Millisecond extraction doesn't work:

```
pymysql.err.ProgrammingError: (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'millisecond FROM t0.timestamp_col) AS tmp \nFROM (SELECT t1.`index` AS `index`, t' at line 1")
```

it seems MySQL doesn't have `millisecond` extraction, but as it has `microsecond` extraction, it can be used with `% 1000`

### PR DESCRIPTION
**Pull request by xmnlab**
Fix #2167 
Fix #2166 
Fix #2168
Fix #2169

### CODE REVIEW

**Comment by xmnlab** [comment]
This PR is done for review. thanks!

File: docs/source/release.rst [hunk 1]
```diff
@@ -13,6 +13,7 @@ Release Notes
 * :feature:`2126` Add translation rules for isnull() and notnull() for pyspark backend
 * :feature:`2062` Implement read_csv for omniscidb backend
 * :feature:`2097` Date, DateDiff and TimestampDiff implementations for OmniSciDB
+* :bug:`2170` Fix millisecond issue for OmniSciDB, MySQL, PostgreSQL and Pandas backends
```
Comment (jreback) [inline docs/source/release.rst:16]: can you add all of the issues numbers here (same issue if fine)

Comment (xmnlab) [reply docs/source/release.rst]: I added all the issues number near each backend name.

File: docs/source/release.rst [hunk 2]
```diff
@@ -13,6 +13,7 @@ Release Notes
 * :feature:`2126` Add translation rules for isnull() and notnull() for pyspark backend
 * :feature:`2062` Implement read_csv for omniscidb backend
 * :feature:`2097` Date, DateDiff and TimestampDiff implementations for OmniSciDB
+* :bug:`2170` Fix millisecond issue for OmniSciDB #2167, MySQL #2169, PostgreSQL #2166 and Pandas #2168 backends
```
Comment (jreback) [inline docs/source/release.rst:16]: add then the same as ``:bug:`2170` `` otherwise these don't render

**Comment by xmnlab** [comment]
test output:

```
11:04 $ IBIS_TEST_MYSQL_PORT=3307 pytest -vv -k millisecond ibis/tests/all/test_temporal.py::test_timestamp_extract
============================ test session starts =============================
platform linux -- Python 3.7.6, pytest-5.3.5, py-1.8.1, pluggy-0.13.0 -- /home/xmn/miniconda3/envs/ibis-dev/bin/python
cachedir: .pytest_cache
rootdir: /home/xmn/dev/quansight/ibis-project/ibis, inifile: setup.cfg
plugins: mock-2.0.0, cov-2.8.1, xdist-1.31.0, forked-1.1.2
collected 91 items / 78 deselected / 13 selected                             

ibis/tests/all/test_temporal.py::test_timestamp_extract[BigQuery-millisecond] SKIPPED [  7%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Clickhouse-millisecond] XFAIL [ 15%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Csv-millisecond] PASSED [ 23%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[HDF5-millisecond] SKIPPED [ 30%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Impala-millisecond] PASSED [ 38%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[MySQL-millisecond] PASSED [ 46%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[OmniSciDB-millisecond] PASSED [ 53%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Pandas-millisecond] PASSED [ 61%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Parquet-millisecond] PASSED [ 69%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[PostgreSQL-millisecond] PASSED [ 76%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[PySpark-millisecond] XFAIL               [ 84%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[SQLite-millisecond] XFAIL                [ 92%]
ibis/tests/all/test_temporal.py::test_timestamp_extract[Spark-millisecond] XFAIL                 [100%]

================== 7 passed, 2 skipped, 78 deselected, 4 xfailed in 85.39s (0:01:25) ===================
```

File: docs/source/release.rst (hunk 2 above)
Comment (xmnlab) [reply docs/source/release.rst]: done! I used `:issue:` for that. thanks!

File: ibis/pandas/execution/temporal.py [hunk 3]
```diff
@@ -33,15 +33,20 @@ def execute_extract_timestamp_field_timestamp(op, data, **kwargs):
     return getattr(data, field_name)
 
 
+@execute_node.register(ops.ExtractTemporalField, pd.Series)
```
Comment (xmnlab) [inline ibis/pandas/execution/temporal.py:36]: `execute_extract_timestamp_field_series` was moved up to be close to above function `execute_extract_timestamp_field_timestamp`

**Comment by jreback** [comment]
thanks @xmnlab

**Pull request merged**